docs/deployment-guide/migrating.md

```diff
@@ -0,0 +1,87 @@
+
+   ```shell
+   kubectl get crds | grep "ngrok" | awk '{print $1}' | xargs -I '{}' kubectl annotate crd '{}' 'meta.helm.sh/release-name=ngrok-operator' --overwrite
+   ```
+   ```shell
+   kubectl get crds | grep "ngrok" | awk '{print $1}' | xargs -I '{}' kubectl annotate crd '{}' 'meta.helm.sh/release-namespace=ngrok-operator' --overwrite
+   ```
+   ```shell
+   kubectl get ingressclass | grep "ngrok" | awk '{print $1}' | xargs -I '{}' kubectl annotate ingressclass '{}' 'meta.helm.sh/release-name=ngrok-operator' --overwrite
+   ```
+   ```shell
+   kubectl get ingressclass | grep "ngrok" | awk '{print $1}' | xargs -I '{}' kubectl annotate ingressclass '{}' 'meta.helm.sh/release-namespace=ngrok-operator' --overwrite
+   ```
+
+   If you are using the gateway feature, you will also need to change the ownership of the gatewayclasses similarly:
+
+   ```shell
+   kubectl get gatewayclasses.gateway.networking.k8s.io | grep "gateway" | awk '{print $1}' | xargs -I '{}' kubectl annotate gatewayclasses.gateway.networking.k8s.io '{}' 'meta.helm.sh/release-name=ngrok-operator' --overwrite
+   ```
+   ```shell
+   kubectl get gatewayclasses.gateway.networking.k8s.io | grep "gateway" | awk '{print $1}' | xargs -I '{}' kubectl annotate gatewayclasses.gateway.networking.k8s.io '{}' 'meta.helm.sh/release-namespace=ngrok-operator' --overwrite
+   ```
+
+4. Install the ngrok Operator chart, with only the `agent` enabled. This will start the new agent to keep your existing applications on-line while you migrate. If you have customized the installation, you may need to adjust the `--set` flags to match your configuration:
+   
+   ```shell
+   helm upgrade ngrok-operator \
+                ngrok/ngrok-operator \
+                --install \
+                --atomic \
+                --wait \
+                --timeout 20m \
+                --namespace ngrok-operator \
+                --set credentials.secret.name=ngrok-credentials \
+                --set replicaCount=0 \
+                --set agent.replicaCount=3
+   ```
+
+5. Scale the old ngrok Ingress Controller deployment to 0 replicas:
+
+   ```shell
+   kubectl -n ngrok-ingress-controller scale deployment <deployment-name> --replicas=0
+   ```
+
+   This will stop the old Ingress Controller from reconciling resources. Verify that your applications are still accessible.
+
+6. Assuming you have no other workloads running in the namespace, you can delete the old ngrok Ingress Controller namespace:
+
+   ```shell
+   kubectl delete namespace ngrok-ingress-controller
+   ```
+
+   If you have other workloads running in the namespace, you will need to manually delete the old resources.
+
+7. Scale the manager replica count to 1:
+
+    ```shell
+    kubectl -n ngrok-operator scale deployment ngrok-operator-manager --replicas=1
+    ```
+
+    Don't forget to update the `replicaCount` in any future helm commands to match.
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines +83 to +88]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Q. Can we double check that this process will not create a kubectl-last-applied annotation that may mess up future deployments if `.spec.replicaCount` is not passed to the `kind: Deployment` template?

Happy to just chat through this offline as a thought exercise to confirm.<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/ngrok/ngrok-operator/446?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
Yeah I was worried about that while reading, but I see now that the replicas are always passed so no problemo.